### PR TITLE
[EVM] add endpoint to return fee history

### PR DESCRIPTION
**Pull request by codchen**
## Describe your changes and provide context
Add `feeHistory` endpoints that returns base fee and rewards percentile (weighted by gas used) for the requested block range. More details on the percentile calculation can be found in comments.

## Testing performed to validate your change
unit tests

### CODE REVIEW

File: evmrpc/info.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	GasUsedRatio []float64        `json:"gasUsedRatio"`
+}
+
+// lastBlock is inclusive
+func (i *InfoAPI) FeeHistory(ctx context.Context, blockCount math.HexOrDecimal64, lastBlock rpc.BlockNumber, rewardPercentiles []float64) (*FeeHistoryResult, error) {
+	result := FeeHistoryResult{}
+
+	// validate reward percentiles
+	for i, p := range rewardPercentiles {
+		if p < 0 || p > 100 || (i > 0 && p < rewardPercentiles[i-1]) {
```
Comment (philipsu522) [inline evmrpc/info.go:42]: - should these be <= and >= respectively?
- do we allow for duplicates?
- do we allow integers only or floats are ok? we can validate here as well

File: evmrpc/info.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+		return int(a.reward) - int(b.reward)
+	})
+	res := []*hexutil.Big{}
+	if len(gasAndRewards) == 0 {
+		return res
+	}
+	var txIndex int
+	sumGasUsed := gasAndRewards[0].gasUsed
+	for _, p := range rewardPercentiles {
+		thresholdGasUsed := uint64(float64(totalEVMGasUsed) * p / 100)
```
[the remark this answers is not in the document]
Comment (philipsu522) [reply evmrpc/info.go:135]: should we use float32 just to be safe?

**Review by philipsu522 — APPROVED** [review]

File: evmrpc/info.go (hunk 1 above)
Comment (codchen) [reply evmrpc/info.go:42]: these checks are copied from https://github.com/ethereum/go-ethereum/blob/master/eth/gasprice/feehistory.go#L227-L232 so to preserve compatibility i think we should just keep them as-is?

**Comment by codchen** [comment]
@philipsu522 ^that's also copied from go-etherem https://github.com/sei-protocol/sei-chain/pull/1106#discussion_r1364377900 Also why would float32 be safer?

**Pull request merged**